pkg/proc/gdbserial/gdbserver.go

```diff
@@ -885,7 +841,6 @@ func (p *Process) Restart(pos string) error {
 
 	p.exited = false
 
-	p.common.ClearAllGCache()
```

Comment:
I think this is needed, in some form.

Reply:
Added a `Restart` method to `(*Target)` which clears the cache and calls `Process.Restart`.

Reply:
What do you think about separating the interface that the backend provides to pkg/proc from the interface that can be called outside of proc, like this: https://github.com/aarzilli/delve/commit/2a2a54467899ca116fef2158679eda29ceaa28b8

ContinueOnce, SetSelectedGoroutine and methods for writing and removing breakpoints would also go inside proc.ProcessInternal.

Reply:
That's an interesting thought, but I want to see how a few more iterations play out first, because I think the embedding of `Process` into `Target` might end up being temporary.

How do you feel about merging as is and revisiting this suggestion?

Reply:
Sure, go ahead.